src/agent/src/device.rs

```diff
@@ -369,6 +402,37 @@ async fn virtio_nvdimm_device_handler(
     update_spec_device_list(device, spec, devidx)
 }
 
+fn split_vfio_option(opt: &str) -> Option<(&str, &str)> {
```

Comment:
Could you add a unit test for this function? Ideally, the other new functions would also have some, but I suspect that might not be particularly easy?

Reply:
I'll see what I can do when I get back to this.  At the moment this PR is on hold while trying to figure out what to do about all the problems of SHPC, or this stuff isn't really usable.

Reply:
ok, thanks @dgibson.

Reply:
Ok, I've added a test for the next spin.